joblib/_parallel_backends.py

```diff
@@ -330,7 +348,8 @@ def _get_pool(self):
         call to apply_async.
         """
         if self._pool is None:
-            self._pool = ThreadPool(self._n_jobs)
+            self._pool = ThreadPool(
+                self._n_jobs, initializer=self.limit_clib_threads)
```

Comment:
I don't think threads have isolated environment variables. Dynamic / ctypes based calls would be required to limit clib threads while joblib is doing thread-based parallelism.

Reply:
I would just not register any initializer at the `ThreadPool` level for now.

Reply:
I confirm that there is a single environment namespace per-process: all threads of the same process share the same environment variables:

```python
>>> from time import sleep
>>> import os
>>> from threading import get_ident
>>> from multiprocessing.pool import ThreadPool
>>> def env_tweaks(*args):
...     tid = str(get_ident())
...     os.environ['THREAD_ID'] = tid
...     sleep(0.01)
...     return (tid, os.environ['THREAD_ID'])
... 
... 
>>> p = ThreadPool(4)
>>> results = p.map(env_tweaks, range(10))
>>> results
[('140286315460352', '140286290282240'), ('140286307067648', '140286315460352'), ('140286298674944', '140286290282240'), ('140286290282240', '140286307067648'), ('140286315460352', '140286298674944'), ('140286307067648', '140286315460352'), ('140286290282240', '140286307067648'), ('140286298674944', '140286307067648'), ('140286315460352', '140286307067648'), ('140286307067648', '140286307067648')]
>>> 

```

Reply:
Indeed.. thanks for pointing that out!